test/disjointCoarseMallocPool.cpp

```diff
@@ -0,0 +1,610 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// This file contains tests for UMF provider API
+
+#include <random>
+#include <set>
+#include <string>
+#include <unordered_map>
+
+#include "../src/memory_provider_internal.h"
+#include "provider.hpp"
+#include "test_helpers.h"
+
+#include <umf/pools/pool_disjoint.h>
+#include <umf/providers/provider_coarse.h>
+#include <umf/providers/provider_os_memory.h>
+
+using umf_test::test;
+
+// TODO move malloc provider somewhere
+
+constexpr int PAGE_SIZE = 4 * 1024;
+
+static enum umf_result_t mallocInitialize(void *params, void **pool) {
+    (void)params;
+    *pool = NULL;
+    return UMF_RESULT_SUCCESS;
+}
+
+static void mallocFinalize(void *pool) { (void)pool; }
+
+static enum umf_result_t mallocAlloc(void *provider, size_t size,
+                                     size_t alignment, void **ptr) {
+    (void)provider;
+    (void)alignment;
+    *ptr = calloc(1, size);
+    return UMF_RESULT_SUCCESS;
+}
+
+static enum umf_result_t mallocFree(void *provider, void *ptr, size_t size) {
+    (void)provider;
+    (void)size;
+    free(ptr);
+    return UMF_RESULT_SUCCESS;
+}
+
+static void mallocGetLastError(void *provider, const char **ppMsg,
+                               int32_t *pError) {
+    (void)provider;
+    (void)ppMsg;
+    (void)pError;
+    assert(0);
+}
+
+static enum umf_result_t
+mallocGetRecommendedPageSize(void *provider, size_t size, size_t *pageSize) {
+    (void)provider;
+    (void)size;
+    *pageSize = PAGE_SIZE;
+    return UMF_RESULT_SUCCESS;
+}
+
+static enum umf_result_t mallocGetPageSize(void *provider, void *ptr,
+
+                                           size_t *pageSize) {
+    (void)provider;
+    (void)ptr;
+    *pageSize = PAGE_SIZE;
+    return UMF_RESULT_SUCCESS;
+}
+
+static enum umf_result_t mallocPurgeLazy(void *provider, void *ptr,
+                                         size_t size) {
+    (void)provider;
+    (void)ptr;
+    (void)size;
+    assert(0);
+    return UMF_RESULT_SUCCESS;
+}
+
+static enum umf_result_t mallocPurgeForce(void *provider, void *ptr,
+                                          size_t size) {
+    (void)provider;
+    (void)ptr;
+    (void)size;
+    assert(0);
+    return UMF_RESULT_SUCCESS;
+}
+
+static enum umf_result_t mallocAllocSplit(void *provider, void *ptr,
+                                          size_t totalSize, size_t firstSize) {
+    (void)provider;
+    (void)ptr;
+    (void)totalSize;
+    (void)firstSize;
+
+    return UMF_RESULT_SUCCESS;
+}
+
+static enum umf_result_t mallocAllocMerge(void *provider, void *ptr1,
+                                          void *ptr2, size_t totalSize) {
+    (void)provider;
+    (void)ptr1;
+    (void)ptr2;
+    (void)totalSize;
+
+    return UMF_RESULT_SUCCESS;
+}
+
+static const char *mallocName(void *provider) {
+    (void)provider;
+    return "malloc";
+}
+
+struct umf_memory_provider_ops_t UMF_MALLOC_MEMORY_PROVIDER_OPS = {
```

Comment:
You don't have to implement this provider - we already have one: https://github.com/oneapi-src/unified-memory-framework/blob/main/test/common/provider.hpp#L113

If you need to customize the `provider_malloc`  you can also just create a struct that inherits from `provider_base_t ` and overrides the functions that you need.

Reply:
Fixed in https://github.com/oneapi-src/unified-memory-framework/pull/715